packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1068,8 +1077,35 @@ Future<void> refreshPluginsList(
   }
 }
 
+/// Injects plugins found in `pubspec.yaml` into the platform-specific projects
+/// only at build-time.
+///
+/// This method is similar to [injectPlugins], but used only for platforms where
+/// the plugin files are not required at compile-time (currently: Web).
```

Comment:
Don't you mean not required at project generation time? If not, I'm not sure what the distinction between "compile time" and "build time" is.

Reply:
Yeah, I think I'm going to use "project creation time" and "build time", I can't think of better names :/

Reply:
I also specified a little bit better what the *actual* difference between the two methods is:

* one writes the generated plugin registrants in locations configured by each `FlutterProject` subclass,
* the other writes the generated plugin registrant (later) and within the temp directory of the build in progress (maybe to a real filesystem, or to an in-memory one)